### PR TITLE
add ros2 control hardware

### PR DESCRIPTION
**Pull request by Mt-is-at-O**
- adapt to ros2_control

### CODE REVIEW

File: mgs1600gy_interface/src/packet_pool.cpp
Comment (Mt-is-at-O) [inline mgs1600gy_interface/src/packet_pool.cpp:1]: I fixed this.

File: mgs1600gy_interface/src/packet_pool.cpp [hunk 1]
```diff
@@ -149,6 +149,8 @@ std::string PacketPool::packetTypeToString(
       return "MZ";
     case PACKET_TYPE::ANG:
       return "ANG";
+    case PACKET_TYPE::GY:
+      return "GY";
```
Comment (Mt-is-at-O) [inline mgs1600gy_interface/src/packet_pool.cpp:153]: I fixed this.

**Comment by Mt-is-at-O** [comment]
`ros2 launch mgs1600gy_bringup hardware.launch.py dev:=`
This command above can launch ros2_control mgs1600gy hardware.

File: mgs1600gy_control/src/mgs1600gy_sensor.cpp [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+    return CallbackReturn::ERROR;
+  }
+
+  SENSOR_MIN_ = std::stof(info.hardware_parameters.at("sensor_min"));
+  SENSOR_MAX_ = std::stof(info.hardware_parameters.at("sensor_max"));
+  NAME_ = info.hardware_parameters.at("name");
+  BASE_LINK_ = NAME_ + "_link";
+  MAGNET_LINK_ = NAME_ + "_magnet_link";
+
+  // TODO : set parameters
```
Comment (Mt-is-at-O) [inline mgs1600gy_control/src/mgs1600gy_sensor.cpp]: mgs1600gy_node gets parameters by using param.yaml which is written in launch file.
But this new harware file can't get parameters such as "roll" or "orientation_covariance" through param.yaml.
So this file needs changes to set up parameters.

File: mgs1600gy_interface/src/packet_pool.cpp (hunk 1 above)
Comment (m12watanabe1a) [reply mgs1600gy_interface/src/packet_pool.cpp:153]: Thank you so much

File: mgs1600gy_control/src/mgs1600gy_sensor.cpp (hunk 2 above)
Comment (m12watanabe1a) [reply mgs1600gy_control/src/mgs1600gy_sensor.cpp]: I've got it.
It should be written in the controller.yaml files.

File: mgs1600gy_interface/include/mgs1600gy_interface/utils.hpp [hunk 3]
```diff
@@ -25,5 +25,8 @@ class Utils
 {
 public:
   static void convertBGR(const std::array<float, 16> &, cv::Mat *, const float, const float);
+  static void convertBRG(
+    const std::array<float, 16> &, std::array<float, 16 * 3> *, const float,
```
Comment (m12watanabe1a) [inline mgs1600gy_interface/include/mgs1600gy_interface/utils.hpp]: `std::array<float, 16 * 3> *` -> `std::array<float, 16 * 3>  &` solves the problem.

**Review by m12watanabe1a — CHANGES_REQUESTED** [review]

**Comment by Mt-is-at-O** [comment]
@m12watanabe1a 
I finished implementation.

**Review by m12watanabe1a — APPROVED** [review]
LGTM

**Pull request merged**